### DISCUSSION TITLE
antd 4.x 版本升级到 antd 5.x 版本过程中，使用Less Loader未能正确读取

### DISCUSSION THREAD

**Discussion by hexiaomin496** [ant-design/ant-design#38772] [Q&A]
在umi项目中, config/config.ts项目中：
 
import { theme } from 'antd';
import { convertLegacyToken } from '@ant-design/compatible';

const { defaultAlgorithm, defaultSeed } = theme;

const mapToken = defaultAlgorithm(defaultSeed);
const v4Token = convertLegacyToken(mapToken);

// Webpack Config
{
  loader: "less-loader",
  options: {
    lessOptions: {
      modifyVars: v4Token,
    },
  },
}

怎么在umi4项目正常使用？

**Comment by zombieJ** [comment]
提供一下 github repo 瞅瞅？

**Reply by hexiaomin496** [reply]
项目是企业级项目，抱歉无法提供私人github代码仓库。

目前 我们在umi4的项目中，在 config/config.ts 文件
<img width="380" alt="image" src="https://user-images.githubusercontent.com/23149314/203793568-5e5c85ea-67b7-480a-ad52-65a1159b8f0d.png">
<img width="361" alt="image" src="https://user-images.githubusercontent.com/23149314/203793641-41ac660c-7422-435a-a654-4ba86d6baa0d.png">
<img width="262" alt="image" src="https://user-images.githubusercontent.com/23149314/203793494-66ff23aa-0b8b-4adc-b2ac-d644a0f4fd46.png">

这样去引用 antd v5的官方文档中 由v4升级到v5的方法，但是此处报了：cannot use import statement outside a module 的错误。

请问在umi4项目中，如何正确地升级到v5，且能兼容以前的less变量呢？请大神指点~感谢！

**Reply by arvinxx** [reply]
升级到最新版antd就好了

**Reply by spt110** [reply]
> 项目是企业级项目，抱歉无法提供私人github代码仓库。
> 
> 目前 我们在umi4的项目中，在 config/config.ts 文件 <img alt="image" width="380" src="https://user-images.githubusercontent.com/23149314/203793568-5e5c85ea-67b7-480a-ad52-65a1159b8f0d.png"> <img alt="image" width="361" src="https://user-images.githubusercontent.com/23149314/203793641-41ac660c-7422-435a-a654-4ba86d6baa0d.png"> <img alt="image" width="262" src="https://user-images.githubusercontent.com/23149314/203793494-66ff23aa-0b8b-4adc-b2ac-d644a0f4fd46.png">
> 
> 这样去引用 antd v5的官方文档中 由v4升级到v5的方法，但是此处报了：cannot use import statement outside a module 的错误。
> 
> 请问在umi4项目中，如何正确地升级到v5，且能兼容以前的less变量呢？请大神指点~感谢！

怎么结局的的，我也遇到同样的问题

**Comment by riverdone** [comment]
您好，请问这个问题现在解决了吗？我升v5项目也报错了：
![image](https://user-images.githubusercontent.com/51936110/204717534-9a157c5e-636c-4ab1-8e9f-af75456fcc33.png)

**Reply by zombieJ** [reply]
这是两个问题，PageHeader 挪到 ProComponent 里了：https://ant-design.antgroup.com/docs/react/migration-v5-cn#%E7%BB%84%E4%BB%B6%E9%87%8D%E6%9E%84%E4%B8%8E%E7%A7%BB%E9%99%A4

**Reply by riverdone** [reply]
我页面中没用PageHeader，用的是@ant-design/pro-layout里的PageContainer

**Reply by riverdone** [reply]
我升级pro-layout版本后，没报PageHeader的错了：
![image](https://user-images.githubusercontent.com/51936110/204755898-89ccfe40-c90a-4c8d-81dc-73e367955372.png)

**Reply by YqxLzx** [reply]
请问解决了吗 我都哭泣了

**Comment by Joeynaive** [comment]
在 defineConfig 中添加配置

lessLoader: {
   modifyVars: v4Token,
},

就行了

**Reply by riverdone** [reply]
![image](https://user-images.githubusercontent.com/51936110/205230839-c5856e1f-0c75-44ea-9165-85109358e1b0.png)
配置了，还是没用呀

**Reply by Joeynaive** [reply]
你的主色不是 #1677ff 吗
<img width="1312" alt="image" src="https://user-images.githubusercontent.com/106252598/205251896-a201150e-b856-4e11-8fc1-a253c9487461.png">

**Reply by riverdone** [reply]
不是#1677ff，但这个报错跟主题色有关吗？

**Reply by Joeynaive** [reply]
抱歉，没看到你报的那个错，那个错是因为5版本去除了导出的less文件，你看global.less文件中或项目中其他文件有没有
 @import '~antd/es/style/themes/default.less'; 有的话去掉就行了

**Reply by riverdone** [reply]
我代码历史没有引入@import '~antd/es/style/themes/default.less'，是.umi缓存文件里引入的：
<img width="1955" alt="image" src="https://user-images.githubusercontent.com/51936110/205256028-01991a2d-7367-4adf-903a-e81cd5860deb.png">
虽然把它注释调掉项目可以正常运行，但页面样式会有问题。

**Reply by Joeynaive** [reply]
删除.umi和node_modules、package-lock.json，重新 npm install，或者看看哪个依赖中使用了这个文件，升级下依赖版本，5版本是调整了一些样式

**Reply by riverdone** [reply]
好的，我先试试，多谢解答哈

**Reply by yangda666** [reply]
hello，请问问题解决了吗，我也遇到类似的问题，项目如果使用的的umi低版本 使用umi中的layout 插件就会出这个错，

**Reply by xiaogonggong-w** [reply]
> 好的，我先试试，多谢解答哈

哈咯,你解决这个问题了么

**Reply by YqxLzx** [reply]
低版本？你用的是umi3吗？

**Reply by YqxLzx** [reply]
hello ， 你解决了吗。。。痛苦死我了
